util.py

```diff
@@ -1540,6 +1543,12 @@ def call(url, *args, **kwargs):
     # stream to short circuit on too-long response bodies (below)
     kwargs.setdefault('stream', True)
 
+    if kwargs.get('headers') is None:
+      kwargs['headers'] =  {}
+
+    if ('user-agent' not in[key.lower() for key in kwargs.get('headers',{}).keys()]):
```

Comment:
good instinct! luckily requests does this for us, so we don't actually need to do it ourselves.